// @tableland/evm/typechain-types/factories/contracts/test/TestTablelandControllerV0.sol/ITablelandControllerV0Test__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ITablelandControllerV0Test__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        inputs: [
            {
                internalType: "address",
                name: "caller",
                type: "address",
            },
        ],
        name: "getPolicy",
        outputs: [
            {
                components: [
                    {
                        internalType: "bool",
                        name: "allowInsert",
                        type: "bool",
                    },
                    {
                        internalType: "bool",
                        name: "allowUpdate",
                        type: "bool",
                    },
                    {
                        internalType: "bool",
                        name: "allowDelete",
                        type: "bool",
                    },
                    {
                        internalType: "string",
                        name: "whereClause",
                        type: "string",
                    },
                    {
                        internalType: "string",
                        name: "withCheck",
                        type: "string",
                    },
                    {
                        internalType: "string[]",
                        name: "updatableColumns",
                        type: "string[]",
                    },
                ],
                internalType: "struct ITablelandControllerV0Test.Policy",
                name: "",
                type: "tuple",
            },
        ],
        stateMutability: "payable",
        type: "function",
    },
];
class ITablelandControllerV0Test__factory {
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
ITablelandControllerV0Test__factory.abi = _abi;
exports.ITablelandControllerV0Test__factory = ITablelandControllerV0Test__factory;
