// @tableland/evm/typechain-types/factories/contracts/test/TestTablelandControllerV0.sol/ITablelandControllerV0Test__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Contract, Signer, utils } from "ethers";
import type { Provider } from "@ethersproject/providers";
import type {
  ITablelandControllerV0Test,
  ITablelandControllerV0TestInterface,
} from "../../../../contracts/test/TestTablelandControllerV0.sol/ITablelandControllerV0Test";

const _abi = [
  {
    inputs: [
      {
        internalType: "address",
        name: "caller",
        type: "address",
      },
    ],
    name: "getPolicy",
    outputs: [
      {
        components: [
          {
            internalType: "bool",
            name: "allowInsert",
            type: "bool",
          },
          {
            internalType: "bool",
            name: "allowUpdate",
            type: "bool",
          },
          {
            internalType: "bool",
            name: "allowDelete",
            type: "bool",
          },
          {
            internalType: "string",
            name: "whereClause",
            type: "string",
          },
          {
            internalType: "string",
            name: "withCheck",
            type: "string",
          },
          {
            internalType: "string[]",
            name: "updatableColumns",
            type: "string[]",
          },
        ],
        internalType: "struct ITablelandControllerV0Test.Policy",
        name: "",
        type: "tuple",
      },
    ],
    stateMutability: "payable",
    type: "function",
  },
] as const;

export class ITablelandControllerV0Test__factory {
  static readonly abi = _abi;
  static createInterface(): ITablelandControllerV0TestInterface {
    return new utils.Interface(_abi) as ITablelandControllerV0TestInterface;
  }
  static connect(
    address: string,
    signerOrProvider: Signer | Provider
  ): ITablelandControllerV0Test {
    return new Contract(
      address,
      _abi,
      signerOrProvider
    ) as ITablelandControllerV0Test;
  }
}
