// @tableland/evm/typechain-types/factories/contracts/policies/ERC721AQueryablePolicies__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
import { Signer, utils, Contract, ContractFactory, Overrides } from "ethers";
import type { Provider, TransactionRequest } from "@ethersproject/providers";
import type { PromiseOrValue } from "../../../common";
import type {
  ERC721AQueryablePolicies,
  ERC721AQueryablePoliciesInterface,
} from "../../../contracts/policies/ERC721AQueryablePolicies";

const _abi = [
  {
    inputs: [],
    name: "ERC721AQueryablePoliciesUnauthorized",
    type: "error",
  },
] as const;

const _bytecode =
  "0x602d6037600b82828239805160001a607314602a57634e487b7160e01b600052600060045260246000fd5b30600052607381538281f3fe73000000000000000000000000000000000000000030146080604052600080fdfea164736f6c6343000813000a";

type ERC721AQueryablePoliciesConstructorParams =
  | [signer?: Signer]
  | ConstructorParameters<typeof ContractFactory>;

const isSuperArgs = (
  xs: ERC721AQueryablePoliciesConstructorParams
): xs is ConstructorParameters<typeof ContractFactory> => xs.length > 1;

export class ERC721AQueryablePolicies__factory extends ContractFactory {
  constructor(...args: ERC721AQueryablePoliciesConstructorParams) {
    if (isSuperArgs(args)) {
      super(...args);
    } else {
      super(_abi, _bytecode, args[0]);
    }
  }

  override deploy(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ERC721AQueryablePolicies> {
    return super.deploy(overrides || {}) as Promise<ERC721AQueryablePolicies>;
  }
  override getDeployTransaction(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): TransactionRequest {
    return super.getDeployTransaction(overrides || {});
  }
  override attach(address: string): ERC721AQueryablePolicies {
    return super.attach(address) as ERC721AQueryablePolicies;
  }
  override connect(signer: Signer): ERC721AQueryablePolicies__factory {
    return super.connect(signer) as ERC721AQueryablePolicies__factory;
  }

  static readonly bytecode = _bytecode;
  static readonly abi = _abi;
  static createInterface(): ERC721AQueryablePoliciesInterface {
    return new utils.Interface(_abi) as ERC721AQueryablePoliciesInterface;
  }
  static connect(
    address: string,
    signerOrProvider: Signer | Provider
  ): ERC721AQueryablePolicies {
    return new Contract(
      address,
      _abi,
      signerOrProvider
    ) as ERC721AQueryablePolicies;
  }
}
